src/hooks/useENSRegistrationStepHandler.tsx

```diff
@@ -166,16 +192,27 @@ export default function useENSRegistrationStepHandler(observer = true) {
           registrationParameters?.commitTransactionConfirmedAt ||
             msBlockTimestamp
         );
-        if (secs > ENS_SECONDS_WAIT) setReadyToRegister(true);
+        if (
+          (secs > ENS_SECONDS_WAIT_WITH_PADDING &&
+            secondsSinceCommitConfirmed > ENS_SECONDS_WAIT_WITH_PADDING) ||
+          // sometimes the provider.getBlock('latest) takes a long time to update to newest block
+          secondsSinceCommitConfirmed > ENS_SECONDS_WAIT_PROVIDER_PADDING
```

Comment:
i was able to see the `provider.getBlock('latest)` was getting stuck for more than 60 secs, so we'd want to skip these cases